Azure iot ca

andrew-buckley opened the pull request:
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

troydai committed:
Add activity log alert commands.

andrew-buckley committed:
Adds support for X.509 Certificates in IoT Hub.

derekbekoe committed:
Remove example metadata for extension (#4785)

iluican commented on src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_help.py:
```diff
@@ -20,6 +23,89 @@
     short-summary: (PREVIEW) Manage Azure IoT hubs.
 """
 
+helps['iot hub certificate'] = """
```
Did you go over this verbiage with a PM (Eustace/Chrissie)?

andrew-buckley replied:
I did not, I'll add him.

andrew-buckley replied:
@eustacea

kisantia committed:
Adding migration flag to sql vnet commands (#4726)

lmazuel committed:
Remove some mock in ACS tests (#4792)

tjprescott committed:
Fix #4745. (#4791)

tjprescott committed:
Update command authoring guidance (#4784)

* Add table and completer guidance. Update --ids guidance.

* Code review feedback.

* Code review feedback.

Commit:
Web: Add --docker-container-logging param (#4718)

williexu committed:
better error messages for `deployment user set` (#4794)

* usererrors

* used yugang's exception handler

* throw exception instead of logging

* refactored handler to command instead of custom method

eustacea commented on src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_help.py:
```diff
@@ -5,6 +5,9 @@
 from azure.cli.core.help_files import helps
 
+certificate_help = ('For a detailed explanation of CA certificates in Azure IoT Hub, '
+                    'see https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-security-x509-get-started')
```
Reference top level document instead: https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-x509ca-overview

eustacea commented on src/command_modules/azure-cli-iot/HISTORY.rst:
```diff
@@ -3,6 +3,10 @@
 
+0.1.15
+++++++
+* Adds support for X.509 certificates.
```
Adds support for certificate authorities (CA) and certificate chains.

eustacea reviewed (commented):
Approved contingent on two minor comment updates.

andrew-buckley committed:
Modifies release notes and changes help link.

derekbekoe committed:
‘azdev build’ for Debian, RPM, Docker, Python wheels, Homebrew (#4750)

* ‘azdev build’ for Debian, RPM, Docker, Python wheels, Homebrew

Use entry points instead of custom scripts

* Fix pylint

derekbekoe committed:
Extensions: support adding by name and list-available (#4788)

* Extensions: support adding by name and list-available

- Update extension publishing readme with new info
- Implement 'az extension list-available'
- Support adding an extension by name
- Define logic for extension update
- Support --index hidden param
- Explicitly only support non platform specific whls
- Add tests

* Minor change

yugangw-msft committed:
vm: support "vm secret add/delete/show" (#4776)

troydai committed:
Update flake8 setting and fix style violations

* Remove E126 escape
* Remove E741 escape
* Annotate the .flake8 file
* Fixes style violation surfaced from escape removal

williexu committed:
Fixed storage metrics update bug (#4819)

yugangw-msft committed:
vm/vmss: expose image plan related arguments (#4783)

vermashi committed:
vm encryption: expose Encrypt-Format-All and Extension publisher/name options (#4765)

ro-joowan committed:
[ADLA - ADLS] - Object refactor (#4812)

panchagnula committed:
Enable zip deployment for webapps and functionapps #4180 (#4700)

djyou committed:
[ACR] Add location in creating webhooks (#4824)

noelbundick committed:
Update Service Fabric _help.py (#4805)

Update example to match entry from arguments list

williexu committed:
Increased block size in blob batch upload command and call upload_blob to reduce duplicate code (#4820)

* moved upload_blob to blob.py, upload-batch calls upload_blob

* fixed mock to point at correct _update_progress, style

* history

* minor fixes-spelling, named param, etc

* reverted determinant name

yugangw-msft committed:
cloud shell: handle the case that 'oid' could be missing in a token (#4828)

derekbekoe committed:
Cloud registration: Prevent users from registering clouds that have missing endpoints (#4830)

* Cloud registration: Prevent users from registering clouds that have missing endpoints

If these endpoints are missed, the CLI fails to load.

* Change version and modify HISTORY

ahmelsayed committed:
App service: support linux function app (#4798)

andrew-buckley committed:
Cleans up .gitignore, and HISTORY.rst. Passes a single factory in commands.py. Moves test utilities. Adds a file completer.

Commit:
BatchAI. Additional command line args to address users' feedback (#4835)

* BatchAI. Added short opt for VM size in file-server create command

Added '-s' short opt for VM size in file-server create command to be consistent with cluster create command.

* BatchAI. Added storage account name and key arguments for cluster creation command.

* BatchAI. Bumped version number and updated HISTORY.rst

* BatchAI. Fixed help for cluster auto-scale

* BatchAI. Addressing CR feedback

1. Removed duplicated parameter
2. Renamed account-name to storage-account-name and account-key to storage-account-key

* BatchAI. Ignoring too many arguments error for cluster create command

* BatchAI. Switched version back because 0.1.3 is not released yet

* BatchAI. Disabled too-many-locals pylint warning

andrew-buckley committed:
Updates style after rebase based on additional linter restrictions.

Commit:
BatchAI. Storage account key and name options (#4841)

* BatchAI. Storage account key and name options

Make --storage-account-name and --storage-account-key work for Azure File share and Azure Storage container specified on command line.

* BatchAI. Fix Uts

williexu committed:
Updated configuring_your_machine doc to not reference scripts by path (#4844)

williexu committed:
'clone root' for consistency (#4845)
